core/src/server/rpc_module.rs

```diff
@@ -1057,7 +1051,10 @@ impl SubscriptionSink {
 
 impl Drop for SubscriptionSink {
 	fn drop(&mut self) {
-		if self.is_active_subscription() {
+		// Subscription was never accepted.
+		if let Some(id) = self.id.take() {
```

Comment:
right, the subscription is only added in `subscribers` if accept has been called successfully?

Reply:
Yep, `accept` adds the `uniq_sub` to the `self.subscribers` on success. 
However, if the `self.id` was not consumed, neither the `accept` nor `reject` calls were called. Submitting `InvalidParams` in that case keeps backwards compatibility with `PendingSubscription`.
Have moved the comment inside the if statement